infrastructure/templates/public-api/components/privateDnsZone.bicep

```diff
@@ -0,0 +1,43 @@
+@description('Specifies the type of zone to create')
+@allowed([
+  'sites'
+  'postgresqlServer'
+])
+param zoneType string
+
+@description('Specifies the resource id of the VNet that this DNS Zone will be attached to')
+@minLength(0)
+param vnetId string
+
+var zoneTypeToNames = {
+  sites: 'privatelink.azurewebsites.net'
+  postgresqlServer: 'privatelink.postgres.database.azure.com'
+}
+
+var privateLinkDnsZoneName = zoneTypeToNames[zoneType]
+
+@description('A DNS zone in which internal DNS records can be managed.')
+resource privateDnsZone 'Microsoft.Network/privateDnsZones@2020-06-01' = {
+  name: privateLinkDnsZoneName
+  location: 'global'
+  properties: {}
+}
+
+@description('A link which makes the internal DNS records within the DNS zone available to other resources on the VNet.')
+resource privateDnsZoneLink 'Microsoft.Network/privateDnsZones/virtualNetworkLinks@2020-06-01' = {
+  parent: privateDnsZone
+  name: '${privateLinkDnsZoneName}-link'
```

Comment:
Yeah that's right.  I don't think there's any hard and fast naming convention for virtual network links - it could be as simple as `vnetlink` or as complex as you like, so long as it's unique within the private dns zone resource which is its parent.  `${privateLinkDnsZoneName}-link` was one such example that I saw when scouring the net for examples which would make it easily identifiable in a list, but I'm happy to change to whatever.

Reply:
Okay cool. Just didn't know if this needed more consistent naming like all our other resources using the standard `s101...` naming conventions. If you're happy with it, fine to leave it!

Reply:
As it goes, when deploying this I twigged that a private DNS zone can be linked to multiple VNets and so it made sense to include the VNet name in the link name to make each VNet link unique.

Rather than going with something too full on like `<vnet-name>-<dns-zone-name>-link` I instead chose a simpler `<vnet-name>-dns-vnet-link`.  I think that in general you'll only see lists of vnet links within the private DNS zone's blades anyway.

Reply:
Nope on second thoughts, I am going for broke and using `<vnet-name>-<dns-zone-name>-vnetlink` to be totally clean on what you're looking at at any given time!